processing_period: 0.05 # seconds
prediction_period: 0.05 # seconds
min_updates_to_confirm: 2

# parameters of the target, specified in the competition rules
ball_wire_length: 1.45 # metres
ball_mode: 0 # 0 - fast ball, 1 - slow ball
ball_speeds: [
  8.0, # metres per second
  5.0, # metres per second
  ]

# parameters for the RANSAC line fitting
linefit:
  fitting_period: 0.2 # seconds
  threshold_distance: 1.2 # point must lie within this distance from the line to be considered inlier during fitting
  min_points: 10
  back_up: 2.0 # metres - how much to back up from the end of the line
  snap_distance: 0.0 # metres
  snap_angle: 0.0 # radians
  preffered_angle: 3.14 # the line with the direction in the XY plane closer to this angle will be chosen
  max_point_ages: [
    50.0, # seconds
    65.0, # seconds
    ]

# parameters of the linear kalman filter
lkf:
  use_acceleration: false # whether to include acceleration in the system model
  min_init_points: 5
  max_speed_err: 3.5 # metre per second
  prediction_horizon: 4.0 # seconds
  prediction_step: 0.4 # seconds
  init_history_duration: 20.0 # seconds
  max_time_since_update: 3.0 # seconds

  process_std:
    position: 0.05
    velocity: 0.01
    acceleration: 0.5

  init_std:
    velocity: 0.5
    acceleration: 0.5

# parameters of the plane-fitting algorithm
rheiv:
  fitting_period: 0.5 # seconds
  max_line_points_ratio: 0.65
  min_points: 10
  max_points: 100
  visualization_size: 100 # metres
  timeout: 1.0 # seconds
  line_threshold_distance: 1.0 # point must lie within this distance from the line to be considered inlier during fitting
  snap_distance: 2.0 # metres
  max_time_since_update: 666.0 # seconds
  
# measurements outside these bounds will be ignored
bounds:
  z:
    min: 5  # metres
    max: 15 # metres
